### ISSUE TITLE
This code is not idempotent

### ISSUE THREAD

**Issue by gudata** [bertvv/ansible-role-samba#6]
This code is not idempotent

```

- name: Create Samba users if they don't exist yet
  shell: >
    (pdbedit -L | grep {{ item.name }} 2>&1 > /dev/null) \
    || (echo {{ item.password }}; echo {{ item.password }}) \
    | smbpasswd -s -a {{ item.name }}
  with_items: samba_users
  when: samba_users is defined
  tags: samba
```

There should be some check if all users in the list exists

```
- name: Check if user exists
  shell: pdbedit -L | grep -c {{ samba_user }} || true
  register: shell_output
  changed_when: False
```
 but with_items ...

**Comment by bertvv** [comment]
True, this is something I wanted to work on, but I haven't gotten to it yet. PR Welcome, though!

**Comment by bertvv** [comment]
Fixed in version 2.1.1
